src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -99,7 +99,18 @@ class MockAppDelegate : public ExchangeDelegate
         {
             ec->GetReliableMessageContext()->SetAckPending(false);
         }
-        ec->Close();
+        if (!mRetainExchange)
+        {
+            ec->Close();
+            ec = nullptr;
+
+            if (mExchange != nullptr)
+            {
+                mExchange->Close();
+                mExchange = nullptr;
```

Comment:
This seems a little weird to me.  If we got a past message with mRetainExchange set to true and this is coming in on the same exchange, `mExchange` and `ec` will be the same object.  Should we avoid closing `mExchange` if `mExchange == ec`?  Or can that not happen?  In the latter case, worth a comment.

Reply:
I'll rework the code to make it cleaner.

Reply:
reworked the code to handle the mismatched exchange condition.